feature-libs/customer-ticketing/core/facade/customer-ticketing.service.ts

```diff
@@ -1,6 +1,73 @@
 import { Injectable } from '@angular/core';
+import {
+  Query,
+  QueryNotifier,
+  QueryService,
+  RoutingService,
+  UserIdService,
+} from '@spartacus/core';
+import {
+  CustomerTicketingFacade,
+  GetTicketQueryReloadEvent,
+  GetTicketQueryResetEvent,
+  TicketDetails,
+} from '@spartacus/customer-ticketing/root';
+import { combineLatest, Observable } from 'rxjs';
+import { map, switchMap, take } from 'rxjs/operators';
+import { CustomerTicketingConnector } from '../connectors';
 
-@Injectable({
-  providedIn: 'root',
-})
-export class CustomerTicketingService {}
+@Injectable()
+export class CustomerTicketingService implements CustomerTicketingFacade {
+  /**
+   * Returns the reload events for the getTicket query.
+   */
+  protected getTicketQueryReloadEvents(): QueryNotifier[] {
+    return [GetTicketQueryReloadEvent];
+  }
+  /**
+   * Returns the reset events for the getTicket query.
+   */
+  protected getTicketQueryResetEvents(): QueryNotifier[] {
+    return [GetTicketQueryResetEvent];
+  }
+
+  protected getTicketQuery$: Query<TicketDetails | undefined> =
+    this.queryService.create<TicketDetails | undefined>(
+      () =>
+        this.checkTicketPreConditions().pipe(
+          switchMap(([customerId, ticketId]) =>
+            this.customerTicketingConnector.getTicket(customerId, ticketId)
+          )
+        ),
+      {
+        reloadOn: this.getTicketQueryReloadEvents(),
+        resetOn: this.getTicketQueryResetEvents(),
+      }
+    );
+
+  constructor(
+    protected queryService: QueryService,
+    protected userIdService: UserIdService,
+    protected customerTicketingConnector: CustomerTicketingConnector,
+    protected routingService: RoutingService
+  ) {}
+
+  protected checkTicketPreConditions(): Observable<[string, string]> {
+    return combineLatest([
+      this.userIdService.getUserId(),
+      this.routingService.getParams(),
+    ]).pipe(
+      take(1),
+      map(([userId, routingParams]) => {
+        if (!userId || !routingParams) {
+          throw new Error('Customer tickets pre conditions not met');
+        }
+        return [userId, routingParams.ticketCode];
+      })
+    );
+  }
+
+  getTicket(): Observable<TicketDetails | undefined> {
+    return this.getTicketQuery$.getState().pipe(map((state) => state.data));
```

Comment:
would you like to expose the 'loading state' and data? If so, no need to map to data

If you would like to just expose the data, then call the .get() function instead of getState() from the query

----
My question is geared towards how would you use it and if it would benefit the customer.
Will you have a spinner upon landing in the details page (flexible as this is not what's happening in other details page, but it should have it)?

Reply:
If you would like to use the state and data, then the type will be of Observable<QueryState<TicketDetails | undefined>> instead

Reply:
yes, I kept the state earlier, but there was no use of it so I changes it to this. 
let's just use the get method instead

Reply:
let's get in a call

Reply:
-------
reminder

UX/UI:
- ask for if there should be a spinner after visiting the details page / refresh the details page
- ask what should happen when a support ticket does not exist (should it rely redirect back to the listing page with a glboal message alert?)

Additional review from your team